sdk/formrecognizer/ai-form-recognizer/src/transforms.ts

```diff
@@ -341,57 +364,127 @@ export function toRecognizeContentResultResponse(
   }
 }
 
-export function toFormModelResponse(response: GetCustomModelResponse): FormModelResponse {
-  const common = {
-    ...response.modelInfo,
-    trainingDocuments: response.trainResult?.trainingDocuments.map((d) => {
-      if (!Array.isArray(d.errors)) {
-        d.errors = [];
-      }
-      return d;
-    }),
-    errors: response.trainResult?.errors,
-    _response: response._response
-  };
+function flattenTrainingDocuments(
+  original: GetCustomModelResponse
+): TrainingDocumentInfo[] | undefined {
+  if (original.composedTrainResults) {
+    // Composed model, need to zip the training documents into a flat array and add modelID correlation.
+    const mappedResultDocuments = original.composedTrainResults.map((innerResult) =>
+      innerResult.trainingDocuments.map((info) => ({
+        ...info,
+        modelId: innerResult.modelId,
+        errors: info.errors ?? []
+      }))
+    );
 
-  if (response.modelInfo.status !== "ready") {
-    return common;
+    // Fold these together using a simple reduce
+    return mappedResultDocuments.reduce(
+      (prev, next) => Array.prototype.concat.call(prev, next),
+      []
+    );
+  } else if (original.trainResult) {
+    // Normal training scenario with only one trainResult
+    return original.trainResult.trainingDocuments.map((info) => ({
+      ...info,
+      modelId: original.modelInfo.modelId,
+      errors: info.errors ?? []
+    }));
   }
 
-  if (response.trainResult?.averageModelAccuracy || response.trainResult?.fields) {
-    // training with forms and labels, populate from trainingResult.fields
-    const fields: Record<string, CustomFormModelField> = {};
-    for (const f of response.trainResult.fields!) {
-      fields[f.fieldName] = { name: f.fieldName, accuracy: f.accuracy, label: null };
+  return undefined;
+}
+
+function toSubmodelsFromComposedTrainResults(results: TrainResult[]): CustomFormSubmodel[] {
+  const mappedSubmodels = results.map((r) => toSubmodelsFromTrainResultLabeled(r));
+
+  // Flatten the array
+  return mappedSubmodels.reduce((prev, next) => Array.prototype.concat.call(prev, next), []);
+}
+
+function toSubmodelsFromTrainResultLabeled(
```

Comment:
why does it return an array even though it is of length 1 if I understand correctly?

Reply:
Composed models and singular models have the same return type from training, and composed models (model training scenarios broadly) have potentially many submodels. So we just wrap this one in an array.